### ISSUE TITLE
partial search is returning data in different format than 12.0.3

### ISSUE THREAD

**Issue by ranjib** [chef/chef#3047]
Partial search supposed to return data in the format of:
example:
```ruby
query.seacrh(:node, '*:*', filter_result: {'attribute' => ['fqdn']})
[ {"url"=>"http://chef-server-url/nodes/nodename", "data"=>{"attribute"=>"somedata"} } ]
```
When searched in localmode, it returns data in different format:
```ruby
[ {"attribute"=>"somedata"}  ]
```
Its likely to be a bug from chef-zero

**Comment by ranjib** [comment]
This will break chef run for all partial search users (from 12.0.3) /cc @chef/client-core

**Comment by btm** [comment]
@ranjib this only affects Chef 12.1.0+ using chef-zero?

**Comment by ranjib** [comment]
12.1.0+ , local ., as well as client-server mode.

**Comment by jdmundrawala** [comment]
https://github.com/chef/chef/pull/2687
https://github.com/chef/chef/issues/2679

It looks like this change was deliberate.

cc @lamont-granquist @mcquin

**Comment by ranjib** [comment]
this is acceptable in a minor release?  or we are considering this is experimental api?

**Comment by lamont-granquist** [comment]
Yes, we documented what we shipped, and 12.0.0 had bugs in what we shipped, so that was fixed in 12.1.0 to comply with the documentation and the design.  That actually complies with SemVer.
